js/components/graph/Sidebar.js

```diff
@@ -1,173 +1,137 @@
 import React from "react";
 import PropTypes from "prop-types";
-import Focus from "./Focus";
 
 export default class Sidebar extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      contentHidden: true,
-      focusDisabled: false,
-      graphActive: 0,
-      graphName: "",
-      toggled: false
+      hidden: true,
+      courses: [],
+      results: []
     };
   }
 
-  componentWillUpdate(prevProps) {
-    if (prevProps.graphName !== this.state.graphName) {
-      this.setState({ graphName: prevProps.graphName }, () => {
-        this.handleFocusEnabled();
+  /**
+   * If the active course props change, update the rendered courses that are in the sidebar.
```

Comment:
Don't use state to store HTML. Instead, this can just be a method that returns the list of course `div`s. (You might worry that this function would be called at each render, which is true, but we can worry about the efficiency of this later, and solve it in a different way.)